clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -295,7 +305,7 @@ public void appendUnchecked(long offset, Record record) {
      * @param entry The entry to append
      */
     public void append(LogEntry entry) {
```

Comment:
I guess you considered whether `append(LogRecord)` and `append(Record)` should be `appendWithOffset`?